### PR TITLE
Add attribute __BlockReflectionAttribute target

### PR DESCRIPTION
**Pull request by maririos**
Add the attribute __BlockReflectionAttribute to the libraries that have targerGroup as netcore50 or netcore50aot, instead of running FxTransform

### CODE REVIEW

**Comment by maririos** [comment]
cc @weshaggard

File: src/Microsoft.DotNet.Build.Tasks/PackageFiles/Build.Common.targets [hunk 1]
```diff
@@ -185,4 +185,20 @@
       <VSDesignTimeBuild Condition="'$(BuildingInsideVisualStudio)'=='true' and '$(BuildingOutOfProcess)'=='false'">true</VSDesignTimeBuild>
     </PropertyGroup>
   </Target>
+
```
Comment (weshaggard) [inline src/Microsoft.DotNet.Build.Tasks/PackageFiles/Build.Common.targets]: Can you please add a comment about what this is? Also add a similar comment in the cs file.

**Comment by weshaggard** [comment]
Other than adding some comments this looks good.

File: src/Microsoft.DotNet.Build.Tasks/PackageFiles/Build.Common.targets (hunk 1 above)
Comment (maririos) [inline src/Microsoft.DotNet.Build.Tasks/PackageFiles/Build.Common.targets]: Done

Comment (weshaggard) [inline src/Microsoft.DotNet.Build.Tasks/PackageFiles/Build.Common.targets]: No need to mention FxTransform as that is legacy at this point. I would simply add the following comment in both places.

Providing a definition for __BlockReflectionAttribute in an assembly is a signal to the .NET Native toolchain to remove the metadata for all non-public APIs. This both reduces size and disables private reflection on those APIs in libraries that include this. The attribute can also be applied to individual public APIs to similarly block them.

Comment (maririos) [inline src/Microsoft.DotNet.Build.Tasks/PackageFiles/Build.Common.targets]: Thanks Wes. It is more clear now

**Comment by maririos** [comment]
CC: @jhendrixMSFT

**Pull request merged**